src/main/java/seedu/address/model/PshModelManager.java

```diff
@@ -165,14 +166,13 @@ public void setSlot(Slot target, Slot editedSlot) {
      */
     @Override
     public ObservableList<Slot> getFilteredSlotList() {
-        //TODO
-        throw new UnsupportedOperationException("Not implemented yet!");
+        return filteredSlots;
     }
 
     @Override
     public void updateFilteredSlotList(Predicate<Slot> predicate) {
```

Comment:
Can we have multiple predicate here? Otherwise i would have to create another class of SlotPredicate resulting in a lot of duplicate code. Alternatively I can implement a predicate with both name and date but that would mean empty parameters sometimes. (refer to my pr)